filebeat/input/filestream/input.go

```diff
@@ -214,6 +215,11 @@ func (inp *filestream) open(log *logp.Logger, canceler input.Canceler, path stri
 		return nil, err
 	}
 
+	r, err = newParsers(r, parserConfig{maxBytes: inp.readerConfig.MaxBytes}, inp.readerConfig.Parsers)
+	if err != nil {
+		return nil, err
+	}
+
 	r = readfile.NewStripNewline(r, inp.readerConfig.LineTerminator)
 	r = readfile.NewLimitReader(r, inp.readerConfig.MaxBytes)
```

Comment:
In case of the logs input we have at least the NewStripNewline before the other readers. The multiline reader joins the events again using `\n` only, no matter which newline was in the file.

Not sure about the LimitReader, Did we put it before or after the multiline in the logs input?

Reply:
Yes, we did. This is the original order of readers in the `log` input:
1. EncodeReader
2. DockerJSONReader
3. JSONReader
4. StripNewlineReader
5. MultilineReader
6. LimitReader

I would like to add most from the readers above to parsers so it can the ordering of those can be configurable except for `EncodeReader`. WDYT?

Reply:
SGTM